src/python/load.rs

```diff
@@ -0,0 +1,35 @@
+use crate::load::LoadFromTiff;
+use ndarray::Array4;
+use numpy::{IntoPyArray, PyArray4};
+use pyo3::{pymodule, types::PyModule, Bound, PyResult, Python};
+use std::fs::File;
+use tiff::decoder::Decoder;
+
+#[pymodule]
+fn dicom_preprocessing<'py>(_py: Python<'py>, m: &Bound<'py, PyModule>) -> PyResult<()> {
+    #[pyfn(m)]
+    #[pyo3(name = "load_tiff_u8")]
+    fn load_tiff_u8<'py>(py: Python<'py>, path: &str) -> Bound<'py, PyArray4<u8>> {
+        let mut decoder = Decoder::new(File::open(path).unwrap()).unwrap();
+        let array = Array4::<u8>::decode(&mut decoder).unwrap();
```

Comment:
Would it be better to have a single `load_tiff` function with a `match` mechanism?
```rust
let decode_func = match dtype {
    "u8" => Array4::<u8>::decode,
    "u16" => Array4::<u16>::decode,
    "f32" => Array4::<f32>::decode,
    "_" => ... // Error,
}
let array = decode_func(&mut decoder).unwrap();
```

And then in Python...
```python
load_tiff(str(path), dtype="u8")
```

Reply:
I looked into this and it seems nontrivial. First, the match arms would have different dtypes which would be a Rust compile time error. I checked to see if there is some generic `PyArray` type that everything could be wrapped into, but it doesn't look like such a thing exists. There is `PyArrayDyn`, but that is aimed at dynamic dimensions, not data type.

I refactored things a bit so that there is minimal logic in each of the data type specific methods, but it doesn't look like we can get down to a single method at the Rust level. Maybe in the future we can have a Python wrapper that dispatches to the correct binding method.